dashboard/src/api/TreeDetails.tsx

```diff
@@ -0,0 +1,17 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query';
+
+import { TreeDetails } from '@/types/tree/TreeDetails';
+
+import http from './api';
+
+const fetchTreeDetailData = async (treeId: string): Promise<TreeDetails> => {
+  const res = await http.get(`/api/tree/${treeId}`);
+  return res.data;
+};
+
+export const useTreeDetail = (treeId: string): UseQueryResult => {
```

Comment:
```suggestion
export const useTreeDetails = (treeId: string): UseQueryResult<TreeDetails> => {
```

so we can avoid `data as TreeDetailsType` in [TreeDetails](https://github.com/profusion/kernelci-dashboard/pull/96/files#diff-09167f3421bfa1cb05a08c18301287157ac0f463661e8bedf33846f29018e808R23)